File: interstellar-travel.go
package main

import "fmt"

// Create the function fuelGauge() here
func fuelGauge(fuel int) {
  fmt.Printf("You have %d gallons of fuel left!", fuel)
}

// Create the function calculateFuel() here
func calculateFuel(planet string) int {
  var fuel int
  if planet == "Venus" {
    fuel = 300000
  } else if planet == "Mercury" {
    fuel = 500000
  } else if planet == "Mars" {
    fuel = 700000
  } else {
    fuel = 0
  }
  return fuel
}

// Create the function greetPlanet() here
func greetPlanet(planet string) {
  fmt.Printf("Hello passengers, we are currently at %v!\n", planet)
}

// Create the function cantFly() here
func cantFly() {
  fmt.Println("We do not have the available fuel to fly there.")
}

// Create the function flyToPlanet() here
func flyToPlanet(planet string, fuel int) int {
  var fuelRemaining, fuelCost int
  fuelRemaining = fuel
  fuelCost = calculateFuel(planet)
  if fuelRemaining >= fuelCost {
    greetPlanet(planet)
    fuelRemaining -= fuelCost
  } else if fuelCost > fuelRemaining {
    cantFly()
  }
  return fuelRemaining
}

func main() {
  // Test your functions!
  // Create `planetChoice` and `fuel`
  planetChoice := "Mars"
  fuel := 1000000
  // And then liftoff!
  fuel = flyToPlanet(planetChoice, fuel)
  fuelGauge(fuel)

}

File: interstellar-travel.py

# Create the function fuel_gage here
def fuel_gauge(fuel):
    print(f"You have {fuel} gallons of fuel left!")

# Create the function calculate_fuel here
def calculate_fuel(planet):
    fuel = 0
    if planet == "Venus":
        fuel = 300000
    elif planet == "Mercury":
        fuel = 500000
    elif planet == "Mars":
        fuel = 700000
    else:
        fuel = 0
    return fuel

# Create the function greet_planet here
def greet_planet(planet):
    print(f"Hello passengers, we are currently at {planet}!")

# Create the function cant_fly here
def cant_fly():
    print("We do not have the available fuel to fly there.")

# Create the function fly_to_planet here
def fly_to_planet(planet, fuel):
    fuel_remaining, fuel_cost = 0, 0
    fuel_remaining = fuel
    fuel_cost = calculate_fuel(planet)
    if fuel_remaining >= fuel_cost:
        greet_planet(planet)
        fuel_remaining -= fuel_cost
    elif fuel_cost > fuel_remaining:
        cant_fly()
    return fuel_remaining

# Create the main function here
def main():
    planet_choice = "Venus"
    fuel = 0

    fuel = fly_to_planet(planet_choice, fuel)
    fuel_gauge(fuel)



if __name__ == "__main__":
    main()